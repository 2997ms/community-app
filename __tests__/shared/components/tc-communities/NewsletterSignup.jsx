import React from 'react';
import Rnd from 'react-test-renderer/shallow';
import NewsletterSignup from 'components/tc-communities/NewsletterSignup';

const rnd = new Rnd();

test('Snapshot match', () => {
  rnd.render((
    <NewsletterSignup
      title="Sign up for our newsletter"
      text="Don’t miss out on the latest Topcoder IOS challenges and information!"
      imageSrc="/themes/wipro2/subscribe-bg.jpg"
    />
  ));
  expect(rnd.getRenderOutput()).toMatchSnapshot();

  rnd.render((
    <NewsletterSignup
      title="Sign up for our newsletter"
      text="Don’t miss out on the latest Topcoder IOS challenges and information!"
      imageSrc="/themes/wipro2/subscribe-bg.jpg"
      theme={{
        container: 'container',
        content: 'content',
        title: 'title',
        text: 'text',
        form: 'form',
        formEmail: 'formEmail',
        formButton: 'formButton',
      }}
    />
  ));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
