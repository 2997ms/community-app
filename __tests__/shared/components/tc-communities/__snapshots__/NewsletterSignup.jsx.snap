// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot match 1`] = `
<NewsletterSignup
  buttonText="Sing Up"
  emailPlaceholder="Email address"
  imageSrc="/themes/wipro2/subscribe-bg.jpg"
  text="Don’t miss out on the latest Topcoder IOS challenges and information!"
  theme={
    Object {
      "container": "src-shared-components-tc-communities-NewsletterSignup-___style__container___3FbuD",
      "content": "src-shared-components-tc-communities-NewsletterSignup-___style__content___3QiYz",
      "form": "src-shared-components-tc-communities-NewsletterSignup-___style__form___3yW86",
      "formButton": "src-shared-components-tc-communities-NewsletterSignup-___style__formButton___Jve55",
      "formEmail": "src-shared-components-tc-communities-NewsletterSignup-___style__formEmail___3N_Uv",
      "text": "src-shared-components-tc-communities-NewsletterSignup-___style__text___1bA_y",
      "title": "src-shared-components-tc-communities-NewsletterSignup-___style__title___ErlHl",
    }
  }
  title="Sign up for our newsletter"
/>
`;

exports[`Snapshot match 2`] = `
<NewsletterSignup
  buttonText="Sing Up"
  emailPlaceholder="Email address"
  imageSrc="/themes/wipro2/subscribe-bg.jpg"
  text="Don’t miss out on the latest Topcoder IOS challenges and information!"
  theme={
    Object {
      "container": "src-shared-components-tc-communities-NewsletterSignup-___style__container___3FbuD container",
      "content": "src-shared-components-tc-communities-NewsletterSignup-___style__content___3QiYz content",
      "form": "src-shared-components-tc-communities-NewsletterSignup-___style__form___3yW86 form",
      "formButton": "src-shared-components-tc-communities-NewsletterSignup-___style__formButton___Jve55 formButton",
      "formEmail": "src-shared-components-tc-communities-NewsletterSignup-___style__formEmail___3N_Uv formEmail",
      "text": "src-shared-components-tc-communities-NewsletterSignup-___style__text___1bA_y text",
      "title": "src-shared-components-tc-communities-NewsletterSignup-___style__title___ErlHl title",
    }
  }
  title="Sign up for our newsletter"
/>
`;
